s3/src/main/resources/reference.conf

```diff
@@ -76,4 +76,8 @@ akka.stream.alpakka.s3 {
 
   # Custom endpoint url, used for alternate s3 implementations
   # endpoint-url = null
+
+  # Which version of the list bucket api to use. Set to 1 to use the old style version 1 API.
+  # By default the newer version 2 api is used.
+  list-bucket-api-version = 2
```

Comment:
Maybe just call it `api-version` but document that it is only effective for `list-bucket` (if additional methods provide multi-versions in the future they'll have to amend the comment)

Reply:
Isn't it clearer, and contained in a single place to leave it as is?

Reply:
My thinking is that it isn't unlikely that the same scenario happens in the future for next version of AWS apis, or for other operations on S3, and then I'm guessing that you would want to apply the version setting to all such operations, not on a method-per-method basis, no?

Reply:
How would you then support (for example) v2  of list bucket and v3 of upload with a single setting?

Reply:
What is the scenario where you would want to have such fine grained control? I'm thinking that if this turns out to be useful for more operations you'd want all of them to use the other version, no?

This isn't that important though so if you have strong feelings for the current setting name, just keep it.